tests/unit/test-util/test.adb.js

```diff
@@ -622,6 +623,75 @@ describe('utils/adb', () => {
        });
   });
 
+  describe('checkOrCleanArtifacts', () => {
+    const artifactDir = `web-ext-artifacts-${Date.now()}`;
+    function makeFakeArtifact(artifactName, isDirectory) {
+      return {
+        name: artifactName,
+        isDirectory: () => {
+          return isDirectory;
+        },
+      };
+    }
+    const files = [
+      makeFakeArtifact(artifactDir.concat('fake1'), true),
+      makeFakeArtifact(artifactDir.concat('fake2'), true),
+      makeFakeArtifact(artifactDir.concat('fake2'), false),
+      makeFakeArtifact('not-web-ext-artifacts', true),
+    ];
+
+    it('checks old artifacts', async () => {
+      const adb = getFakeADBKit({
+        adbClient: {
+          readdir: sinon.spy(() => Promise.resolve(files)),
+          shell: sinon.spy(() => Promise.resolve('')),
+        },
+        adbkitUtil: {
+          readAll: sinon.spy(() => Promise.resolve(Buffer.from('1\n'))),
+        },
+      });
+      const adbUtils = new ADBUtils({adb});
+
+      const promise = adbUtils.checkOrCleanArtifacts('device1', false);
+      const result = await assert.isFulfilled(promise);
+      assert.equal(result, true);
+
+      sinon.assert.calledOnce(adb.fakeADBClient.readdir);
+      sinon.assert.notCalled(adb.fakeADBClient.shell);
```

Comment:
Nit, let's add an inline comment right above this line to explain what notCalled is meant to assert (just something like `// No file should be removed and so shell shouldn't be called in this test.`)